### PR TITLE
[IDP-1221] Extract schema type from TypeResult signature from endpoints

### PR DESCRIPTION
**Pull request by heqianwang**
## Description of changes
We added a new IOperationFilter to automatically take the schema types from the API endpoints so that users won't have to annotate their API endpoints with `[ProducesResponseType]` or `[SwaggerResponse]` annotations for their APIs. The Operation also does not overwrite these endpoints if it was already defined.

## Breaking changes
None

## Additional checks
Use preview version of library to validate with an API

- [x] Updated the documentation of the project to reflect the changes
- [x] Added new tests that cover the code changes

### CODE REVIEW

File: src/tests/WebApi.OpenAPI.SystemTest/ExtractTypeResult/TypedResultController.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        return TypedResults.Ok(new TypedResultExample("Example"));
+    }
+
+    [HttpGet]
+    [Route("/withNoAnnotation")]
+    [ProducesResponseType(typeof(TypedResultExample), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public Results<Ok<TypedResultExample>, BadRequest<ProblemDetails>, NotFound> TypedResultWithNoAnnotation(int id)
```
Comment (PrincessMadMath) [inline src/tests/WebApi.OpenAPI.SystemTest/ExtractTypeResult/TypedResultController.cs:25]: Should the naming be?
- TypedResultWithAnnotation
- TypedResultWithNoAnnotation

Comment (PrincessMadMath) [reply src/tests/WebApi.OpenAPI.SystemTest/ExtractTypeResult/TypedResultController.cs:25]: (also update the routes name)

File: src/tests/WebApi.OpenAPI.SystemTest/OperationId/OperationIdMinimalApis.cs [hunk 2]
```diff
@@ -1,15 +1,15 @@
-﻿namespace WebApi.OpenAPI.SystemTest.OperationId;
+namespace WebApi.OpenAPI.SystemTest.OperationId;
 
 public static class OperationIdMinimalApis
 {
     public static void AddEndpointsForOperationId(this WebApplication app)
     {
-        app.MapGet("minimal-endpoint-with-name", (() => Results.Ok()))
+        app.MapGet("minimal-endpoint-with-name", (() => TypedResults.Ok()))
```
Comment (PrincessMadMath) [inline src/tests/WebApi.OpenAPI.SystemTest/OperationId/OperationIdMinimalApis.cs:7]: Why the change in the test of OperationId?

File: src/tests/WebApi.OpenAPI.SystemTest/openapi-v1.yaml [hunk 3]
```diff
@@ -42,4 +42,210 @@ paths:
       responses:
         '200':
           description: Success
-components: { }
\ No newline at end of file
+  '/minimal-endpoint-with-typed-result-no-produces/{id}':
```
Comment (PrincessMadMath) [inline src/tests/WebApi.OpenAPI.SystemTest/openapi-v1.yaml:45]: We should almost used our pipeline to detect breaking changes to OpenAPI.spec.

**Review by PrincessMadMath — DISMISSED** [review]

File: src/Workleap.Extensions.OpenAPI/TypedResult/ExtractSchemaTypeResultFilter.cs [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        {
+            yield break;
+        }
+
+        var genericTypeCount = returnType.GenericTypeArguments.Length;
+
+        // For type like Ok, BadRequest, NotFound
+        if (genericTypeCount == 0)
+        {
+            // Exclude raw IResult since we can't infer the status code --> minimal API causing issues here.
```
[the remark this answers is not in the document]
Comment (heqianwang) [reply src/Workleap.Extensions.OpenAPI/TypedResult/ExtractSchemaTypeResultFilter.cs:97]: Is a leftover comment, no longer necessary

File: src/tests/WebApi.OpenAPI.SystemTest/openapi-v1.yaml (hunk 3 above)
Comment (heqianwang) [reply src/tests/WebApi.OpenAPI.SystemTest/openapi-v1.yaml:45]: good point, though, oasdiff pipeline is hosted on ADO and we don't have the equivalent on github reusable workflows

File: src/tests/WebApi.OpenAPI.SystemTest/ExtractTypeResult/TypedResultController.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    [SwaggerResponse(StatusCodes.Status200OK, "Returns TypedResult", typeof(TypedResultExample), "application/json")]
+    public Ok<ProblemDetails> TypedResultWithSwaggerResponseAnnotation()
+    {
+        return TypedResults.Ok(new ProblemDetails());
+    }
+
+    [HttpGet]
+    [Route("/producesResponseTypeAnnotation")]
+    [ProducesResponseType(typeof(TypedResultExample), StatusCodes.Status200OK)]
+    public Ok<ProblemDetails> TypedResultWithProducesResponseTypeAnnotation()
```
[the remark this answers is not in the document]
Comment (heqianwang) [reply src/tests/WebApi.OpenAPI.SystemTest/ExtractTypeResult/TypedResultController.cs:66]: It was to see if the annotation overrides the Result<> type, combined into the other test

File: src/Workleap.Extensions.OpenAPI/TypedResult/ExtractSchemaTypeResultFilter.cs [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+                    continue;
+                }
+            }
+
+            var response = new OpenApiResponse();
+            if (responseMetadata.SchemaType != null)
+            {
+                var schema = context.SchemaGenerator.GenerateSchema(responseMetadata.SchemaType, context.SchemaRepository);
+
+                var contentTypes = GetContentTypes(context);
```
[the remark this answers is not in the document]
Comment (heqianwang) [reply src/Workleap.Extensions.OpenAPI/TypedResult/ExtractSchemaTypeResultFilter.cs:39]: Hmm I am not sure if we should move this to the top? given its only used right below, the context is fresh when reading the code

**Review by PrincessMadMath — APPROVED** [review]
